File: app/src/main/java/ru/merkulyevsasha/gosduma/presentation/deputies/DeputiesView.java
package ru.merkulyevsasha.gosduma.presentation.deputies;


import java.util.List;

import ru.merkulyevsasha.gosduma.models.Deputy;
import ru.merkulyevsasha.gosduma.presentation.MvpView;

interface DeputiesView extends MvpView {

    void prepareToSearch(String searchText);

    void showData(List<Deputy> items);
    void showDataEmptyMessage();

    void showDeputyDetailsScreen(Deputy deputy);

    void showFilterDialog(List<Integer> filter);
    void showSortDialog(int currentItemIndex);

}

File: deputies/src/main/java/ru/merkulyevsasha/deputies/DeputiesView.kt
package ru.merkulyevsasha.deputies

import ru.merkulyevsasha.coreandroid.base.BaseView
import ru.merkulyevsasha.gdcore.models.Deputy

interface DeputiesView : BaseView {
    fun showItems(items: List<Deputy>)
    fun showProgress()
    fun hideProgress()
}

